lib/iris/tests/idiff.py

```diff
@@ -139,24 +139,26 @@ def skip(event):
 
 
 def _calculate_hit(uris, phash, action):
+    # Extract the hex basename strings from the uris.
+    hexes = [os.path.splitext(os.path.basename(uri))[0] for uri in uris]
     # Create the expected perceptual image hashes from the uris.
     to_hash = imagehash.hex_to_hash
-    expected = [to_hash(os.path.splitext(os.path.basename(uri))[0],
-                        hash_size=iris.tests._HASH_SIZE)
-                for uri in uris]
+    expected = [to_hash(uri_hex) for uri_hex in hexes]
     # Calculate the hamming distance vector for the result hash.
     distances = [e - phash for e in expected]
+
     if action == 'first':
         index = 0
     elif action == 'last':
         index = -1
     elif action == 'similar':
         index = np.argmin(distances)
-    elif action == 'difference':
+    elif action == 'different':
```

Comment:
😕. What's happening here?

Reply:
It was **always** wrong! See https://github.com/SciTools/iris/blob/master/lib/iris/tests/idiff.py#L288